### PR TITLE
Add Tensorflow Object Detection API.

### PR DESCRIPTION
**Pull request by derekjchow**
For details see our paper:
"Speed/accuracy trade-offs for modern convolutional object detectors."
Huang J, Rathod V, Sun C, Zhu M, Korattikara A, Fathi A, Fischer I,
Wojna Z, Song Y, Guadarrama S, Murphy K, CVPR 2017
https://arxiv.org/abs/1611.10012

### CODE REVIEW

File: object_detection/README.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+<a href='g3doc/configuring_jobs.md'>
+  Configuring an object detection pipeline</a><br>
+<a href='g3doc/preparing_inputs.md'>Preparing inputs</a><br>
+<a href='g3doc/running_locally.md'>Running locally</a><br>
+<a href='g3doc/running_on_cloud.md'>Running on the cloud</a><br>
+<a href='g3doc/detection_model_zoo.md'>Tensorflow detection model zoo</a><br>
+<a href='g3doc/exporting_models.md'>
+  Exporting a trained model for inference</a><br>
+<a href='g3doc/defining_your_own_model.md'>
+  Defining your own model architecture</a><br>
```
Comment (sguada) [inline object_detection/README.md]: Add a link to the notebook

Comment (derekjchow) [inline object_detection/README.md]: The top link provides instructions for the notebook.

File: object_detection/builders/hyperparams_builder.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+  Returns:
+    A dictionary containing batch_norm parameters.
+  """
+  batch_norm_params = {
+      'decay': batch_norm.decay,
+      'center': batch_norm.center,
+      'scale': batch_norm.scale,
+      'epsilon': batch_norm.epsilon,
+      'is_training': is_training and batch_norm.train,
```
Comment (derekjchow) [inline object_detection/builders/hyperparams_builder.py]: Done.

File: object_detection/anchor_generators/BUILD [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+licenses(["notice"])
+
+# Apache 2.0
+py_library(
+    name = "grid_anchor_generator",
+    srcs = [
+        "grid_anchor_generator.py",
+    ],
+    deps = [
+        "//pyglib",
```
Comment (derekjchow) [inline object_detection/anchor_generators/BUILD]: Nope. Removed.

File: object_detection/README.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+* Jonathan Huang, github: [jch1](https://github.com/jch1)
+* Vivek Rathod, github: [tombstone](https://github.com/tombstone)
+* Derek Chow, github: [derekjchow](https://github.com/derekjchow)
+* Chen Sun, github: [jesu9](https://github.com/jesu9)
+* Menglong Zhu, github: [dreamdragon](https://github.com/dreamdragon)
+
+
+## Table of contents
+
```
Comment (sguada) [inline object_detection/README.md]: Replace with

Quick Start:
 - Jupyter notebook for off-the-shelf inference
 - Training on a pet detector

Installation
 - Requirements
 - Configuring an object detection pipeline
 - Preparing inputs

Running:
 - Locally
 - On the cloud

Extras:
 - Tensorflow detection model zoo
 - Exporting a trained model for inference
 - Defining your own model architecture

File: object_detection/README.md (hunk 1 above)
Comment (sguada) [inline object_detection/README.md]: no it is a link to the .md file not the notebook itself

File: object_detection/README.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+computer vision needs, and we hope that you will as well.
+<p align="center">
+  <img src="g3doc/img/kites_detections_output.jpg" width=676 height=450>
+</p>
+Contributions to the codebase are welcome and we would love to hear back from
+you if you find this API useful.  Finally if you use the Tensorflow Object
+Detection API for a research publication, please consider citing:
+
+```
+"Speed/accuracy trade-offs for modern convolutional object detectors."
```
Comment (sguada) [inline object_detection/README.md]: Add link to arxiv paper and bibtex

File: object_detection/g3doc/running_notebook.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+To run the Jupyter notebook, run the following command from
+`tensorflow/models/object_detection`:
+
+```
+# From tensorflow/models/object_detection
+jupyter notebook
+```
+
+The notebook should open in your favorite web browser. Click the
+`object_detection_tutorial.ipynb` link to open the demo.
```
Comment (sguada) [inline object_detection/g3doc/running_notebook.md]: Add link to the notebook itself, github can show it.

Also add the instructions on how to run it to the notebook itself

File: object_detection/g3doc/installation.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+* Protobuf 2.6
+* Pillow 1.0
+* lxml
+* tf Slim (which is included in the "tensorflow/models" checkout)
+* Jupyter notebook
+* Matplotlib
+* Tensorflow
+
+To install Tensorflow, follow the
+[Tensorflow installation instructions](https://www.tensorflow.org/install/).
```
Comment (sguada) [inline object_detection/g3doc/installation.md]: Could you provide a couple of lines for a simple install of TF and direct to the link for others.

File: object_detection/README.md (hunk 5 above)
Comment (sguada) [inline object_detection/README.md]: https://arxiv.org/abs/1611.10012

File: object_detection/README.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+In addition to our base Tensorflow detection model definitions, this
+release includes:
+
+* A selection of trainable detection models, including:
+  * Single Shot Multibox Detector (SSD) with MobileNet,
+  * SSD with Inception V2,
+  * Region-Based Fully Convolutional Networks (R-FCN) with Resnet 101,
+  * Faster RCNN with Resnet 101,
+  * Faster RCNN with Inception Resnet v2
+  * Mask R-CNN.
```
Comment (sguada) [inline object_detection/README.md]: does it use some architecture ResNet 101?

File: object_detection/README.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+* Frozen weights (trained on the COCO dataset) for each of the above models to be used for out-of-the-box inference purposes.
+* A Jupyter notebook for performing out-of-the-box inference with one of our released models
```
Comment (sguada) [inline object_detection/README.md]: Add link to the notebook

File: object_detection/README.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+* Convenient local training scripts as well as distributed training and evaluation pipelines via Google Cloud
```
Comment (sguada) [inline object_detection/README.md]: Add link to local training and to GC training

File: object_detection/g3doc/detection_model_zoo.md [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+Inside the un-tar'ed directory, you will find:
+
+* a graph proto (`graph.pbtxt`)
+* a checkpoint
+  (`model.ckpt.data-00000-of-00001`, `model.ckpt.index`, `model.ckpt.meta`)
+* a frozen graph proto with weights baked into the graph as constants
+  (`frozen_inference_graph.pb`) to be used for out of the box inference
+    (try this out in the Jupyter notebook!)
```
Comment (sguada) [inline object_detection/g3doc/detection_model_zoo.md]: Add link to the notebook

File: object_detection/g3doc/exporting_models.md [hunk 12]
```diff
@@ -0,0 +1 @@
+# Exporting a trained model for inference
```
Comment (sguada) [inline object_detection/g3doc/exporting_models.md]: This file is empty

File: object_detection/g3doc/running_notebook.md (hunk 6 above)
Comment (derekjchow) [inline object_detection/g3doc/running_notebook.md]: Done.

File: object_detection/README.md (hunk 5 above)
Comment (derekjchow) [inline object_detection/README.md]: Link added. The bibtext seems a bit big for the README.

http://dblp.uni-trier.de/rec/bibtex/journals/corr/HuangRSZKFFWSG016

File: object_detection/README.md (hunk 4 above)
Comment (derekjchow) [inline object_detection/README.md]: Great idea! Done.

File: object_detection/g3doc/detection_model_zoo.md (hunk 11 above)
Comment (derekjchow) [inline object_detection/g3doc/detection_model_zoo.md]: Done.

File: object_detection/g3doc/exporting_models.md (hunk 12 above)
Comment (derekjchow) [inline object_detection/g3doc/exporting_models.md]: Filled in.

File: object_detection/g3doc/detection_model_zoo.md (hunk 11 above)
Comment (derekjchow) [inline object_detection/g3doc/detection_model_zoo.md]: Done.

File: object_detection/README.md (hunk 5 above)
Comment (sguada) [inline object_detection/README.md]: At link to it would be enough
https://scholar.googleusercontent.com/scholar.bib?q=info:l291WsrB-hQJ:scholar.google.com/&output=citation&scisig=AAGBfm0AAAAAWUIIlnPZ_L9jxvPwcC49kDlELtaeIyU-&scisf=4&ct=citation&cd=-1&hl=en&scfhb=1

Comment (derekjchow) [inline object_detection/README.md]: Done.

File: object_detection/anchor_generators/BUILD (hunk 3 above)
Comment (marlazinger200) [inline object_detection/anchor_generators/BUILD]: 1

File: object_detection/builders/hyperparams_builder.py (hunk 2 above)
Comment (marlazinger200) [inline object_detection/builders/hyperparams_builder.py]: 1

File: object_detection/README.md (hunk 1 above)
Comment (marlazinger200) [inline object_detection/README.md]: 1

File: object_detection/g3doc/running_notebook.md (hunk 6 above)
Comment (marlazinger200) [inline object_detection/g3doc/running_notebook.md]: 1

File: object_detection/README.md (hunk 5 above)
Comment (marlazinger200) [inline object_detection/README.md]: 1

File: object_detection/g3doc/installation.md (hunk 7 above)
Comment (marlazinger200) [inline object_detection/g3doc/installation.md]: 1

File: object_detection/g3doc/detection_model_zoo.md (hunk 11 above)
Comment (marlazinger200) [inline object_detection/g3doc/detection_model_zoo.md]: 1

File: object_detection/g3doc/exporting_models.md (hunk 12 above)
Comment (marlazinger200) [inline object_detection/g3doc/exporting_models.md]: 1

**Comment by marlazinger200** [comment]
1
